### ISSUE TITLE
[Bug]: ignoreAttributes for react/jsx-no-leaked-render causes a schema failure

### ISSUE THREAD

**Issue by philipbordallo** [jsx-eslint/eslint-plugin-react#3991]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

Following the docs for [`react/jsx-no-leaked-render`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md) and using [`ignoreAttributes`](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md#ignoreattributes) as intended:
```jsonc
{
  "react/jsx-no-leaked-render": ["warn", { "ignoreAttributes": true }]
}
```

Causes a schema error:
```
Configuration for rule "react/jsx-no-leaked-render" is invalid: Value {"ignoreAttributes":true,"validStrategies":["ternary","coerce"]} should NOT have additional properties.
```

As the last version on npm (`v7.37.5`) seems to be missing the schema definition:

```js
{
  schema: [
    {
      type: 'object',
      properties: {
        validStrategies: {
          type: 'array',
          items: {
            enum: [
              TERNARY_STRATEGY,
              COERCE_STRATEGY,
            ],
          },
          uniqueItems: true,
          default: DEFAULT_VALID_STRATEGIES,
        },
      },
      additionalProperties: false,
    },
  ],
}
```

[See on UNPKG](https://app.unpkg.com/eslint-plugin-react@7.37.5/files/lib/rules/jsx-no-leaked-render.js#L125-144)

Versus:
https://github.com/jsx-eslint/eslint-plugin-react/blob/8501c298f2f50ccd0d343a4c435d29bee3a28288/lib/rules/jsx-no-leaked-render.js#L140-L162

### Expected Behavior

`ignoreAttributes` should exist in the schema definition.

### eslint-plugin-react version

v7.37.5

### eslint version

v8.42.0

### node version

v22.20.0

**Comment by chewmanji** [comment]
@ljharb any plans to make a release in the near future?

**Comment by ljharb** [comment]
The next release will be once we have eslint 10 compat, which is blocked on #3977 and eslint-plugin-import.
